src/app/clusters/sample-mei-server/sample-mei-server.cpp

```diff
@@ -0,0 +1,228 @@
+    auto endpointIndex = EndpointIndex(endpoint);
+    if (endpointIndex == std::numeric_limits<size_t>::max())
+    {
+        ctxt.mCommandHandler.AddStatus(ctxt.mRequestPath, Protocols::InteractionModel::Status::UnsupportedEndpoint);
+        return;
+    }
+
+    switch (ctxt.mRequestPath.mCommandId)
+    {
+    case Commands::Ping::Id:
+        HandleCommand<Commands::Ping::DecodableType>(ctxt, [endpoint](HandlerContext & ctx, const auto & req) {
+            ChipLogProgress(Zcl, "Ping Command on endpoint %d", endpoint);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Success);
+        });
+        return;
+    case Commands::AddArguments::Id:
+        HandleCommand<Commands::AddArguments::DecodableType>(ctxt, [endpoint](HandlerContext & ctx, const auto & req) {
+            ChipLogProgress(Zcl, "AddArgumentsCommand on endpoint %d", endpoint);
+            if (req.arg1 > UINT8_MAX - req.arg2)
+            {
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand);
+                return;
+            }
+
+            AddArgumentsResponse::Type response;
+            response.returnValue = static_cast<uint8_t>(req.arg1 + req.arg2);
+            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+        });
+        return;
+    }
+}
+
+CHIP_ERROR SampleMeiServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err     = CHIP_NO_ERROR;
+    auto endpoint      = aPath.mEndpointId;
+    auto endpointIndex = EndpointIndex(endpoint);
+    if (endpointIndex == std::numeric_limits<size_t>::max())
+    {
+        return CHIP_ERROR_NO_ENDPOINT;
+    }
+
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::FlipFlop::Id:
+        ChipLogProgress(Zcl, "Read Attribute flip-flop from endpoint %d index %u value %d", endpoint,
+                        static_cast<uint16_t>(endpointIndex), content[endpointIndex].flipflop);
+        err = aEncoder.Encode(content[endpointIndex].flipflop);
+        break;
+    default:
+        break;
+    }
+
+    return err;
+}
+
+CHIP_ERROR SampleMeiServer::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    CHIP_ERROR err     = CHIP_NO_ERROR;
+    auto endpoint      = aPath.mEndpointId;
+    auto endpointIndex = EndpointIndex(endpoint);
+    if (endpointIndex == std::numeric_limits<size_t>::max())
+    {
+        return CHIP_ERROR_NO_ENDPOINT;
+    }
+
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::FlipFlop::Id: {
+        auto oldValue = content[endpointIndex].flipflop;
+        ReturnErrorOnFailure(aDecoder.Decode(content[endpointIndex].flipflop));
+        ChipLogProgress(Zcl, "Write Attribute flip-flop on endpoint %d index %u newValue %d oldValue %d", endpoint,
+                        static_cast<uint16_t>(endpointIndex), content[endpointIndex].flipflop, oldValue);
+    }
+    break;
```

Comment:
I would expect this break to be inside the brackets for the case....

Reply:
fixed